code/modules/events/_event.dm

```diff
@@ -120,8 +119,10 @@ Runs the event
 	testing("[time2text(world.time, "hh:mm:ss")] [E.type]")
 	triggering = TRUE
 
-	if (alert_observers && !admin_forced)
-		message_admins("Random Event triggering in [DisplayTimeText(RANDOM_EVENT_ADMIN_INTERVENTION_TIME)]: [name] (<a href='?src=[REF(src)];cancel=1'>CANCEL</a>).")
+	// Make sure the other admins know if it's another admin trying to get an event through, or if the game is engaging in chicanery. We sleep here for the given amount of time to make sure they have enough time to cancel it.
+	// We do a bit of baby-proofing in case an admin fat-fingers an event and summons a Blob instead of something like an Ion Storm.
+	if(alert_observers || admin_forced) // we don't really need to message admins over every tiny event (space dust), this gets the important ones though. We always want to give the admin a way to cancel something they accidentally force.
```

Comment:
why not cancel in prerunevent instead?

Reply:
i picked this one because i thought some events didn't call prerunevent, i can invert the behavior to go back to prerunevent and remove the check on runevent.

Reply:
I also don't know what you mean by un-needed code, it has to go somewhere. If I delete it on PreRunEvent, I have to move it to RunEvent, and vice-versa.

Reply:
i'm moving it to prerunevent anyways though